src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -4,51 +4,197 @@
 
 package frc.robot.subsystems.drive;
 
-import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveDriveOdometry;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj.ADIS16470_IMU;
+import frc.robot.Constants;
+import frc.robot.Constants.DriveConstants;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
-import com.ctre.phoenix.sensors.Pigeon2;
-import com.revrobotics.CANSparkMax;
-import com.revrobotics.CANSparkMaxLowLevel.MotorType;
-
 public class DriveSubsystem extends SubsystemBase {
-  Pigeon2 pigeon = new Pigeon2(1);
-  CANSparkMax spark = new CANSparkMax(2, MotorType.kBrushless);
+  private double targetHeading;
+
+  // Create MAXSwerveModules
+  private final SwerveModule m_frontLeft = new SwerveModule(
+      DriveConstants.FRONT_LEFT_DRIVING_CAN_ID,
+      DriveConstants.FRONT_LEFT_TURNING_CAN_ID,
+      DriveConstants.FRONT_LEFT_CHASSIS_ANGULAR_OFFSET);
```

Comment:
Add a TODO comment that we need to add some angular offsets that are specific to the position of the module. This has to do with how we're setting the angular offset.